File: raincoat-fabric/src/main/java/dev/uten2c/raincoat/model/RaincoatModelReloadListener.java
package dev.uten2c.raincoat.model;

import dev.uten2c.raincoat.RaincoatMod;
import net.fabricmc.fabric.api.resource.SimpleSynchronousResourceReloadListener;
import net.minecraft.resource.ResourceManager;
import net.minecraft.util.Identifier;

public final class RaincoatModelReloadListener implements SimpleSynchronousResourceReloadListener {
    @Override
    public Identifier getFabricId() {
        return new Identifier(RaincoatMod.MOD_ID, "rainmodel");
    }

    @Override
    public void reload(ResourceManager manager) {
        GunUnbakedModel.clearCache();
    }
}

File: raincoat-fabric/src/main/kotlin/dev/uten2c/raincoat/model/RaincoatModelReloadListener.kt
package dev.uten2c.raincoat.model

import dev.uten2c.raincoat.MOD_ID
import net.fabricmc.fabric.api.resource.SimpleSynchronousResourceReloadListener
import net.minecraft.resource.ResourceManager
import net.minecraft.util.Identifier

class RaincoatModelReloadListener : SimpleSynchronousResourceReloadListener {
    override fun getFabricId(): Identifier {
        return Identifier(MOD_ID, "rainmodel")
    }

    override fun reload(manager: ResourceManager) {
        GunUnbakedModel.clearCache()
    }
}
